patroni/ctl.py

```diff
@@ -1253,7 +1253,14 @@ def _do_failover_or_switchover(obj: Dict[str, Any], action: str, cluster_name: s
         raise PatroniCtlException(action.title() + ' target and source are the same.')
 
     if candidate and candidate not in candidate_names:
-        raise PatroniCtlException('Member {0} does not exist in cluster {1}'.format(candidate, cluster_name))
+        raise PatroniCtlException(
+            f'Member {candidate} does not exist in cluster {cluster_name} or is tagged as nofailover')
+
+    if not force and action == 'failover':
+        if global_config.is_synchronous_mode and not cluster.sync.is_empty\
+           and not cluster.sync.matches(candidate, True)\
+           and not click.confirm(f'Are you sure you want to failover to the asynchronous node {candidate}'):
+            raise PatroniCtlException('Aborting ' + action)
```

Comment:
```suggestion
    if not force and action == 'failover':
        if all((
                global_config.is_synchronous_mode,
                not cluster.sync.is_empty,
                not cluster.sync.matches(candidate, True),
                not click.confirm(f'Are you sure you want to failover to the asynchronous node {candidate}')
        )):
            raise PatroniCtlException('Aborting ' + action)
```

Reply:
Or do all the non-interactive checks first:
```suggestion
    if all((
            not force,
            action == 'failover',
            global_config.is_synchronous_mode,
            not cluster.sync.is_empty,
            not cluster.sync.matches(candidate, True),
    )):
        if not click.confirm(f'Are you sure you want to failover to the asynchronous node {candidate}'):
            raise PatroniCtlException('Aborting ' + action)

```